examples/demo-app/src/actions.js

```diff
@@ -305,3 +329,51 @@ export function loadSampleConfigurations(sampleMapId = null) {
   }
 }
 
+/**
+ * this action will be triggered when the file is being uploaded
+ * @param isLoading
+ * @param metadata
+ * @returns {{type: string, isLoading: *, metadata: *}}
+ */
+export function setPushingFile(isLoading, metadata) {
+  return {
+    type: PUSHING_FILE,
+    isLoading,
+    metadata
+  };
+}
+
+/**
+ * This method will export the current kepler config file to the choosen cloud platform
+ * @param data
+ * @param handlerName
+ * @returns {Function}
+ */
+export function exportFileToCloud(data, handlerName = 'dropbox') {
+  const authHandler = AUTH_HANDLERS[handlerName];
+  return dispatch => {
+    // we are exporting to json format with 2 spaces,
+    // we could save bandwidth if we used a single line
+    // but it wouldn't be readable
+    const newBlob = new Blob([JSON.stringify(data, null, 2)], {type: 'application/json'});
+    const file = new File([newBlob], `kepler.gl/keplergl_${(new Date()).toISOString()}.json`);
```

Comment:
maybe we can use a generator to generate padded numbers `001`, `002` so that the file name doesn't get too long. 

```
const pad = num =>  `00${num}`.slice(-3);
const numberGenerator = () => {
  let index = 0;
  return () => pad(index+=1);
}
const getNum = numberGenerator();

getNum() // '001'
getNum() // '002'
getNum() // '003'
```

Reply:
I think this will work for a single session but if users restart the app and save a map the new file name will start again from 001. I need to find something that will work across multiple sessions

Reply:
I updated the logic to generate a unique and short file name by re-using generateHashId
```
kepler.gl/keplergl_${generateHashId(6)}.json
```